history_analyser.py

```diff
@@ -44,15 +45,33 @@
 elif browser.startswith('f'):
     start = time()
     if platform.startswith("win"):
-        history_folder = path.join(Path.home(),"AppData\\Roaming\\Mozilla\\Firefox\\Profiles")
+        profilePath = "AppData\\Roaming\\Mozilla\\Firefox\\Profiles"
+        history_folder = path.join(Path.home(), profilePath)
+
+        print("Profiles available are: ")
+        listdir(profilePath)
     elif platform=="linux":
         history_folder = path.join(Path.home(), ".mozilla/firefox")
     elif platform=="darwin":
-        history_folder = path.join(Path.home(), "Library/Application Support/Firefox/Profiles")
+        profilePath = path.join(Path.home(), "Library/Application Support/Firefox/Profiles/")
+        history_folder = path.join(Path.home(), profilePath)
+
+        print("Profiles available are: ")
+        lis = list(listdir(profilePath))
+        dirList = []
+        for i in lis:
+            for f in listdir(path.join(profilePath, i)):
```

Comment:
Any progress?

Reply:
`i` is for every profile that is present. I go into the profile and check if it contains the `.sqlite` file needed as you said. So the two for loops do this.

Reply:
I understood that the first loop goes through the list of directories, but in the second loop you concatenate `profilePath` and `i` (using the path.join function). So you are looping through all the files in `Profile/1`, `Profile/2`, etc directories which do not exist (even for Chrome, the profile directories are `Profile 1`, `Profile 2`, etc.). 
So you need to change this to loop through all files in the second loop (and not directories).

Reply:
I think that is what it is doing. `i` iterates over all the profiles present in the `Profile/` folder that is give by `lis = list(listdir(profilePath))`. Then `j` goes over the files in the particular profile given by `listdir(path.join(profilePath, i))` and if it contain the `.sqlite` file it appends this profile to the list of profiles that will be displayed to the user to choose from.

Reply:
Sorry for the super late reply!
I understand that you are iterating over the profiles present in `Profile` directory, but `path.join` just concatenates two paths, so `path.join(profilePath, i)` will give `Profile/1`, `Profile/2`, `Profile/3`, etc. Since, in firefox, the profile directories have random names like `Profile/pdq9henv.default`, your loop will not work.